airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/copy/s3/S3StreamCopier.java

```diff
@@ -35,7 +36,9 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(S3StreamCopier.class);
 
-  private static final int DEFAULT_UPLOAD_THREADS = 10; // The S3 cli uses 10 threads by default.
+  // The S3 cli uses 10 threads by default, but each thread will hold
+  // upto partSize amount of memory. So we cannot set it to very high.
+  private static final int DEFAULT_UPLOAD_THREADS = 2;
```

Comment:
I think the default 10 threads is one major root cause of the high memory usage. But I cannot test it now because the Snowflake user is temporarily locked out for whatever reason...